bcda/cli.go

```diff
@@ -101,17 +164,27 @@ func importCCLF9(fileMetadata cclfFileMetadata) error {
 	)
 
 	sc := bufio.NewScanner(file)
+	count := 0
+	validator := cclfFileValidator["CCLF9"]
+
 	for sc.Scan() {
 		b := sc.Bytes()
-		if len(bytes.TrimSpace(b)) > 0 {
+		bytelength := len(bytes.TrimSpace(b))
+		if bytelength > 0 && bytelength <= validator.maxRecordLength {
+			count++
+			// currently only errors if there are more records than we expect.
+			if count > validator.totalRecordCount {
+				return errors.New("maximum record count reached: did not match cclf0 validation")
+			}
 			fmt.Printf("\nXREF: %s\n", b[0:1])
 			fmt.Printf("Current identifier: %s\n", b[currIDStart:currIDEnd])
 			fmt.Printf("Previous identifier: %s\n", b[prevIDStart:prevIDEnd])
 			fmt.Printf("Previous identifier effective date: %s\n", b[prevIDEffDateStart:prevIDEffDateEnd])
 			fmt.Printf("Previous identifier obsolete date: %s\n", b[prevIDObsDateStart:prevIDObsDateEnd])
+		} else {
+			return errors.New("incorrect record length: did not match cclf0 validation")
```

Comment:
Same logging comments as above.  We should try to log as much as possible on failure.

Reply:
the logging happens here: importCCLFDirectory as the error will bubble up.

Reply:
Doesn't look like that function reports the exact file name that failed though.

Reply:
yep this was clarified in slack. On it